plugin/trino-exasol/src/test/java/io/trino/plugin/exasol/TestExasolConnectorTest.java

```diff
@@ -0,0 +1,470 @@
+import static io.trino.spi.type.VarcharType.VARCHAR;
+import static io.trino.testing.MaterializedResult.resultBuilder;
+import static java.lang.String.format;
+import static java.util.Locale.ENGLISH;
+import static java.util.stream.Collectors.joining;
+import static java.util.stream.IntStream.range;
+import static org.assertj.core.api.Assertions.assertThat;
+import static org.assertj.core.api.Assertions.assertThatThrownBy;
+import static org.junit.jupiter.api.Assumptions.abort;
+
+@Isolated
+public class TestExasolConnectorTest
+        extends BaseJdbcConnectorTest
+{
+    private TestingExasolServer exasolServer;
+
+    @Override
+    protected QueryRunner createQueryRunner()
+            throws Exception
+    {
+        exasolServer = closeAfterClass(new TestingExasolServer());
+        return createExasolQueryRunner(
+                exasolServer,
+                ImmutableMap.of(),
+                ImmutableMap.of(),
+                REQUIRED_TPCH_TABLES);
+    }
+
+    @Override
+    protected boolean hasBehavior(TestingConnectorBehavior connectorBehavior)
+    {
+        return switch (connectorBehavior) {
+            // Tests requires write access which is not implemented
+            case SUPPORTS_JOIN_PUSHDOWN,
+                    SUPPORTS_TOPN_PUSHDOWN,
+                    SUPPORTS_AGGREGATION_PUSHDOWN,
+                    SUPPORTS_LIMIT_PUSHDOWN -> false;
+
+            // Parallel writing is not supported due to restrictions of the Exasol JDBC driver.
+            case SUPPORTS_CREATE_SCHEMA,
+                    SUPPORTS_CREATE_TABLE,
+                    SUPPORTS_RENAME_TABLE,
+                    SUPPORTS_INSERT,
+                    SUPPORTS_UPDATE,
+                    SUPPORTS_DELETE,
+                    SUPPORTS_ADD_COLUMN,
+                    SUPPORTS_RENAME_COLUMN,
+                    SUPPORTS_COMMENT_ON_TABLE,
+                    SUPPORTS_SET_COLUMN_TYPE -> false;
+
+            case SUPPORTS_ROW_TYPE,
+                    SUPPORTS_ARRAY,
+                    SUPPORTS_NEGATIVE_DATE // min date is 0001-01-01
+                    -> false;
+
+            default -> super.hasBehavior(connectorBehavior);
+        };
+    }
+
+    @Override
+    protected Optional<DataMappingTestSetup> filterDataMappingSmokeTestData(DataMappingTestSetup setup)
+    {
+        String type = setup.getTrinoTypeName();
+        if (type.equals("time") || type.startsWith("time(")) {
+            // Exasol does not have a TIME type
+            return Optional.of(setup.asUnsupported());
+        }
+        if (type.startsWith("timestamp") && type.endsWith("with time zone")) {
+            // Exasol only supports TIMESTAMP WITH _LOCAL_ TIMEZONE
+            return Optional.of(setup.asUnsupported());
+        }
+        if (type.startsWith("timestamp(6)")) {
+            // Exasol supports timestamps with milliseconds precision, not more.
+            return Optional.of(setup.asUnsupported());
+        }
```

Comment:
the type names are trino type names. So even if Exasol doesn't support timestamp with precision higher than 3 the connector can perform rounding when reading/writing higher precision values.

"writing" is an overloaded term because it also includes things like a timestamp literal in the predicate of a query.

Reply:
How can I add support for this? Do you have an example in another connector?